src/Doctrine/DoctrineHelper.php

```diff
@@ -104,7 +104,13 @@ public function doesClassUseDriver(string $className, string $driverClass): bool
             // if the class should be generated with attributes or annotations. If this exception is thrown, we will check based on the
             // namespaces for the given $className and compare it with the doctrine configuration to get the correct MappingDriver.
 
-            return $this->isInstanceOf($this->getMappingDriverForNamespace($className), $driverClass);
+            $classNameComponents = explode('\\', $className);
+            if (1 < \count($classNameComponents)) {
+                array_pop($classNameComponents);
+            }
+            $classNamespace = implode('\\', $classNameComponents);
```

Comment:
Could you add a comment about which situation this is addressing? A test would be ideal, though I don't know how easy that would be to add.

Thanks!